gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/spring/ServiceConfiguration.java

```diff
@@ -62,7 +62,7 @@
  */
 @Configuration
 @ComponentScan("io.gravitee.rest.api.service")
-@EnableTransactionManagement
+@EnableTransactionManagement(proxyTargetClass = true)
```

Comment:
I had to add this in order to correctly inject service components properly without error.
Otherwise, spring creates proxies for services implementing other interfaces, which causes dependency injection problems.
For now, still not clear to me, and not sure about the impacts.

Reply:
@brasseld any opinion on this?

Reply:
Note: https://docs.spring.io/spring-framework/docs/3.0.0.M4/reference/html/ch07s06.html#:~:text=Spring%20AOP%20uses%20either%20JDK,dynamic%20proxy%20will%20be%20used.

Reply:
Before going further, and even if spring says "there is no much impact", I would recommand to run quick bench test to analyse memories consumption, because we are changing how every bean are gonna be created.

Reply:
I do agree with @guillaumelamirand and I would definitely prefer to understand why this is needed now...

Reply:
With default spring proxying, bean injection leads to this error at startup :
`Bean named 'EntrypointServiceV4' is expected to be of type 'io.gravitee.rest.api.service.v4.EntrypointService' but was actually of type 'com.sun.proxy.$Proxy91'`
Without the interface, Spring handle this bean an an untyped proxy, and so can't import it as a dependency.

This configuration forces the usage of Spring CGLIB proxying, like explained in above documentation.

Here are the comparison of 5 MAPI startup of this branch compared to master :

**Master branch**

  | startup time (ms) | loaded classes | used heap (Mo) | used metaspace (Mo)
-- | -- | -- | -- | --
1 | 2909 | 18889 | 106 | 96
2 | 2985 | 18941 | 118 | 95
3 | 3076 | 18900 | 115 | 95
4 | 2917 | 18873 | 117 | 95
5 | 3020 | 18980 | 124 | 95
AVERAGE | 2981.4 | 18916.6 | 116 | 95.2

**Branch without interfaces**

  | startup time (ms) | loaded classes | used heap (Mo) | used metaspace (Mo)
-- | -- | -- | -- | --
1 | 3576 | 18940 | 105 | 96
2 | 3138 | 18901 | 119 | 96
3 | 3433 | 18891 | 115 | 95
4 | 3328 | 18843 | 117 | 96
5 | 3298 | 18953 | 122 | 96
AVERAGE | 3354.6 | 18905.6 | 115.6 | 95.8

We can notice the increased startup time.